go-controller/pkg/node/gateway_iptables.go

```diff
@@ -78,20 +96,37 @@ func delIptRules(rules []iptRule) error {
 }
 
 func getGatewayInitRules(chain string, proto iptables.Protocol) []iptRule {
-	return []iptRule{
-		{
-			table:    "nat",
-			chain:    "PREROUTING",
-			args:     []string{"-j", chain},
-			protocol: proto,
-		},
-		{
-			table:    "nat",
-			chain:    "OUTPUT",
-			args:     []string{"-j", chain},
-			protocol: proto,
-		},
+	iptRules := []iptRule{}
+	if chain == iptableITPChain {
+		iptRules = append(iptRules,
+			iptRule{
+				table:    "mangle",
+				chain:    "OUTPUT",
+				args:     []string{"-j", chain},
+				protocol: proto,
+			},
+		)
+	} else {
+		iptRules = append(iptRules,
+			iptRule{
+				table:    "nat",
+				chain:    "PREROUTING",
+				args:     []string{"-j", chain},
+				protocol: proto,
+			},
+		)
 	}
+	if chain != iptableETPChain { // ETP chain only meant for external traffic
+		iptRules = append(iptRules,
```

Comment:
why u didn't move this block inside the else {

Reply:
Since ETP chain needs to be called only from PREROUTING and ITP chain needs to be called from OUTPUT I needed to add this check instead of it being inside the else block. If I put it inside else, ITP won't have a jump from OUTPUT.